File: server/emailService.js
const { ClientSecretCredential } = require('@azure/identity');
const { Client } = require('@microsoft/microsoft-graph-client');
const { TokenCredentialAuthenticationProvider } = require('@microsoft/microsoft-graph-client/authProviders/azureTokenCredentials');

class EmailService {
  constructor() {
    this.clientId = process.env.OUTLOOK_CLIENT_ID;
    this.clientSecret = process.env.OUTLOOK_CLIENT_SECRET;
    this.tenantId = process.env.OUTLOOK_TENANT_ID;
    this.senderEmail = process.env.OUTLOOK_SENDER_EMAIL || 'westgatewarehouse@suniquecabinetry.com';
    
    if (!this.clientId || !this.clientSecret || !this.tenantId) {
      console.error('Outlook API credentials are not properly configured in .env file');
      this.isConfigured = false;
    } else {
      this.isConfigured = true;
      this.initializeClient();
    }
  }

  initializeClient() {
    try {
      const credential = new ClientSecretCredential(
        this.tenantId,
        this.clientId,
        this.clientSecret
      );

      const authProvider = new TokenCredentialAuthenticationProvider(credential, {
        scopes: ['https://graph.microsoft.com/.default']
      });

      this.graphClient = Client.initWithMiddleware({
        authProvider: authProvider
      });
    } catch (error) {
      console.error('Error initializing Microsoft Graph client:', error);
      this.isConfigured = false;
    }
  }

  async sendConfirmationEmail(appointmentDetails) {
    if (!this.isConfigured) {
      console.warn('Email service not configured. Skipping email send.');
      return { success: false, message: 'Email service not configured' };
    }

    try {
      const { orderNumber, pickupTime, customerEmail } = appointmentDetails;
      
      if (!customerEmail) {
        return { success: false, message: 'Customer email is required' };
      }
      
      const recipients = [{
        emailAddress: {
          address: customerEmail
        }
      }];
      
      const ccRecipients = [{
        emailAddress: {
          address: 'zackwu204@gmail.com'
        }
      }];

      const emailBody = this.generateEmailBody(orderNumber, pickupTime);
      
      const message = {
        message: {
          subject: `Appointment Confirmation - Order ${orderNumber}`,
          body: {
            contentType: 'HTML',
            content: emailBody
          },
          toRecipients: recipients,
          ccRecipients: ccRecipients
        },
        saveToSentItems: true
      };

      await this.graphClient
        .api(`/users/${this.senderEmail}/sendMail`)
        .post(message);

      console.log('Confirmation email sent successfully');
      return { success: true, message: 'Email sent successfully' };
    } catch (error) {
      console.error('Error sending confirmation email:', error);
      return { success: false, message: error.message };
    }
  }

  async sendCancellationEmail(appointmentDetails) {
    if (!this.isConfigured) {
      console.warn('Email service not configured. Skipping email send.');
      return { success: false, message: 'Email service not configured' };
    }

    try {
      const { orderNumber, appointmentDate, appointmentTime, customerEmail } = appointmentDetails;
      
      if (!customerEmail) {
        return { success: false, message: 'Customer email is required' };
      }
      
      const recipients = [{
        emailAddress: {
          address: customerEmail
        }
      }];
      
      const ccRecipients = [{
        emailAddress: {
          address: 'zackwu204@gmail.com'
        }
      }];

      const emailBody = this.generateCancellationEmailBody(orderNumber, appointmentDate, appointmentTime);
      
      const message = {
        message: {
          subject: `Appointment Cancelled - Order ${orderNumber}`,
          body: {
            contentType: 'HTML',
            content: emailBody
          },
          toRecipients: recipients,
          ccRecipients: ccRecipients
        },
        saveToSentItems: true
      };

      await this.graphClient
        .api(`/users/${this.senderEmail}/sendMail`)
        .post(message);

      console.log('Cancellation email sent successfully');
      return { success: true, message: 'Email sent successfully' };
    } catch (error) {
      console.error('Error sending cancellation email:', error);
      return { success: false, message: error.message };
    }
  }

  async sendRescheduleEmail(appointmentDetails) {
    if (!this.isConfigured) {
      console.warn('Email service not configured. Skipping email send.');
      return { success: false, message: 'Email service not configured' };
    }

    try {
      const { orderNumber, oldAppointmentDate, oldAppointmentTime, newPickupTime, customerEmail } = appointmentDetails;
      
      if (!customerEmail) {
        return { success: false, message: 'Customer email is required' };
      }
      
      const recipients = [{
        emailAddress: {
          address: customerEmail
        }
      }];
      
      const ccRecipients = [{
        emailAddress: {
          address: 'zackwu204@gmail.com'
        }
      }];

      const emailBody = this.generateRescheduleEmailBody(orderNumber, oldAppointmentDate, oldAppointmentTime, newPickupTime);
      
      const message = {
        message: {
          subject: `Appointment Rescheduled - Order ${orderNumber}`,
          body: {
            contentType: 'HTML',
            content: emailBody
          },
          toRecipients: recipients,
          ccRecipients: ccRecipients
        },
        saveToSentItems: true
      };

      await this.graphClient
        .api(`/users/${this.senderEmail}/sendMail`)
        .post(message);

      console.log('Reschedule email sent successfully');
      return { success: true, message: 'Email sent successfully' };
    } catch (error) {
      console.error('Error sending reschedule email:', error);
      return { success: false, message: error.message };
    }
  }

  generateEmailBody(orderNumber, pickupTime) {
    const formattedDateTime = new Date(pickupTime).toLocaleString('en-US', {
      weekday: 'long',
      year: 'numeric',
      month: 'long',
      day: 'numeric',
      hour: '2-digit',
      minute: '2-digit',
      timeZone: 'America/Chicago'
    });

    return `
      <!DOCTYPE html>
      <html>
      <head>
        <style>
          body {
            font-family: Arial, sans-serif;
            line-height: 1.6;
            color: #333;
          }
          .container {
            max-width: 600px;
            margin: 0 auto;
            padding: 20px;
          }
          .header {
            background: linear-gradient(135deg, #515a36 0%, #6b7444 100%);
            color: white;
            padding: 30px 20px;
            text-align: center;
            border-radius: 10px 10px 0 0;
          }
          .header h1 {
            margin: 0;
            font-size: 24px;
          }
          .content {
            background: #ffffff;
            padding: 30px;
            border: 1px solid #e5e7eb;
            border-top: none;
          }
          .details {
            background: #f9fafb;
            padding: 20px;
            border-radius: 8px;
            margin: 20px 0;
          }
          .details p {
            margin: 10px 0;
          }
          .details strong {
            color: #515a36;
          }
          .instructions {
            background: #fff8e1;
            padding: 20px;
            border-left: 4px solid #f59e0b;
            margin: 20px 0;
          }
          .instructions h3 {
            margin-top: 0;
            color: #f59e0b;
          }
          .instructions ol {
            margin: 10px 0;
            padding-left: 20px;
          }
          .instructions li {
            margin: 10px 0;
          }
          .footer {
            background: #f9fafb;
            padding: 20px;
            text-align: center;
            border-radius: 0 0 10px 10px;
            border: 1px solid #e5e7eb;
            border-top: none;
          }
          .footer p {
            margin: 5px 0;
            font-size: 14px;
            color: #6b7280;
          }
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>Appointment Confirmation</h1>
            <p>Sunique Cabinetry Pickup Appointment</p>
          </div>
          <div class="content">
            <p>Your pickup appointment has been confirmed!</p>
            
            <div class="details">
              <p><strong>Order Number:</strong> ${orderNumber}</p>
              <p><strong>Pickup Time:</strong> ${formattedDateTime}</p>
            </div>
            
            <div class="instructions">
              <h3>Important Instructions</h3>
              <ol>
                <li>Please save this confirmation for your reference and prepare it when picking up.</li>
                <li>To cancel or modify your appointment, please contact Sunique at <strong>(972) 245-3309</strong>.</li>
                <li>The appointment is reserved for 10 minutes. If you are delayed more than 10 minutes, the appointment will be void.</li>
                <li>We reserve the right to adjust the time. We will do our best to get your order ready, but we do not guarantee it will be ready at the exact scheduled time.</li>
              </ol>
            </div>
            
            <p>If you have any questions, please don't hesitate to contact us.</p>
          </div>
          <div class="footer">
            <p><strong>Sunique Cabinetry</strong></p>
            <p>Phone: (972) 245-3309</p>
            <p>This is an automated confirmation email.</p>
          </div>
        </div>
      </body>
      </html>
    `;
  }

  generateCancellationEmailBody(orderNumber, appointmentDate, appointmentTime) {
    return `
      <!DOCTYPE html>
      <html>
      <head>
        <style>
          body {
            font-family: Arial, sans-serif;
            line-height: 1.6;
            color: #333;
          }
          .container {
            max-width: 600px;
            margin: 0 auto;
            padding: 20px;
          }
          .header {
            background: linear-gradient(135deg, #dc2626 0%, #ef4444 100%);
            color: white;
            padding: 30px 20px;
            text-align: center;
            border-radius: 10px 10px 0 0;
          }
          .header h1 {
            margin: 0;
            font-size: 24px;
          }
          .content {
            background: #ffffff;
            padding: 30px;
            border: 1px solid #e5e7eb;
            border-top: none;
          }
          .details {
            background: #fee2e2;
            padding: 20px;
            border-radius: 8px;
            margin: 20px 0;
            border-left: 4px solid #dc2626;
          }
          .details p {
            margin: 10px 0;
          }
          .details strong {
            color: #dc2626;
          }
          .info {
            background: #fff8e1;
            padding: 20px;
            border-left: 4px solid #f59e0b;
            margin: 20px 0;
          }
          .footer {
            background: #f9fafb;
            padding: 20px;
            text-align: center;
            border-radius: 0 0 10px 10px;
            border: 1px solid #e5e7eb;
            border-top: none;
          }
          .footer p {
            margin: 5px 0;
            font-size: 14px;
            color: #6b7280;
          }
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>Appointment Cancelled</h1>
            <p>Sunique Cabinetry Pickup Appointment</p>
          </div>
          <div class="content">
            <p>Your pickup appointment has been cancelled.</p>
            
            <div class="details">
              <p><strong>Order Number:</strong> ${orderNumber}</p>
              <p><strong>Cancelled Appointment:</strong> ${appointmentDate} at ${appointmentTime}</p>
            </div>
            
            <div class="info">
              <p><strong>What to do next:</strong></p>
              <p>If you would like to reschedule your pickup appointment, please contact Sunique at <strong>(972) 245-3309</strong> or visit our scheduling system to book a new time slot.</p>
            </div>
            
            <p>We apologize for any inconvenience. If you have any questions, please don't hesitate to contact us.</p>
          </div>
          <div class="footer">
            <p><strong>Sunique Cabinetry</strong></p>
            <p>Phone: (972) 245-3309</p>
            <p>This is an automated notification email.</p>
          </div>
        </div>
      </body>
      </html>
    `;
  }

  generateRescheduleEmailBody(orderNumber, oldAppointmentDate, oldAppointmentTime, newPickupTime) {
    const formattedNewDateTime = new Date(newPickupTime).toLocaleString('en-US', {
      weekday: 'long',
      year: 'numeric',
      month: 'long',
      day: 'numeric',
      hour: '2-digit',
      minute: '2-digit',
      timeZone: 'America/Chicago'
    });

    return `
      <!DOCTYPE html>
      <html>
      <head>
        <style>
          body {
            font-family: Arial, sans-serif;
            line-height: 1.6;
            color: #333;
          }
          .container {
            max-width: 600px;
            margin: 0 auto;
            padding: 20px;
          }
          .header {
            background: linear-gradient(135deg, #2563eb 0%, #3b82f6 100%);
            color: white;
            padding: 30px 20px;
            text-align: center;
            border-radius: 10px 10px 0 0;
          }
          .header h1 {
            margin: 0;
            font-size: 24px;
          }
          .content {
            background: #ffffff;
            padding: 30px;
            border: 1px solid #e5e7eb;
            border-top: none;
          }
          .details {
            background: #eff6ff;
            padding: 20px;
            border-radius: 8px;
            margin: 20px 0;
            border-left: 4px solid #2563eb;
          }
          .details p {
            margin: 10px 0;
          }
          .details strong {
            color: #2563eb;
          }
          .old-time {
            background: #fee2e2;
            padding: 15px;
            border-radius: 5px;
            margin: 10px 0;
            text-decoration: line-through;
            color: #6b7280;
          }
          .new-time {
            background: #dcfce7;
            padding: 15px;
            border-radius: 5px;
            margin: 10px 0;
            color: #16a34a;
            font-weight: bold;
          }
          .instructions {
            background: #fff8e1;
            padding: 20px;
            border-left: 4px solid #f59e0b;
            margin: 20px 0;
          }
          .instructions h3 {
            margin-top: 0;
            color: #f59e0b;
          }
          .instructions ol {
            margin: 10px 0;
            padding-left: 20px;
          }
          .instructions li {
            margin: 10px 0;
          }
          .footer {
            background: #f9fafb;
            padding: 20px;
            text-align: center;
            border-radius: 0 0 10px 10px;
            border: 1px solid #e5e7eb;
            border-top: none;
          }
          .footer p {
            margin: 5px 0;
            font-size: 14px;
            color: #6b7280;
          }
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>Appointment Rescheduled</h1>
            <p>Sunique Cabinetry Pickup Appointment</p>
          </div>
          <div class="content">
            <p>Your pickup appointment has been rescheduled.</p>
            
            <div class="details">
              <p><strong>Order Number:</strong> ${orderNumber}</p>
              
              <div class="old-time">
                <p><strong>Previous Time:</strong> ${oldAppointmentDate} at ${oldAppointmentTime}</p>
              </div>
              
              <div class="new-time">
                <p><strong>New Pickup Time:</strong> ${formattedNewDateTime}</p>
              </div>
            </div>
            
            <div class="instructions">
              <h3>Important Instructions</h3>
              <ol>
                <li>Please save this confirmation for your reference and prepare it when picking up.</li>
                <li>If this time does not work for you, please contact Sunique at <strong>(972) 245-3309</strong> to reschedule.</li>
                <li>The appointment is reserved for 10 minutes. If you are delayed more than 10 minutes, the appointment will be void.</li>
                <li>We reserve the right to adjust the time. We will do our best to get your order ready, but we do not guarantee it will be ready at the exact scheduled time.</li>
              </ol>
            </div>
            
            <p>We apologize for any inconvenience. If you have any questions, please don't hesitate to contact us.</p>
          </div>
          <div class="footer">
            <p><strong>Sunique Cabinetry</strong></p>
            <p>Phone: (972) 245-3309</p>
            <p>This is an automated notification email.</p>
          </div>
        </div>
      </body>
      </html>
    `;
  }
}

module.exports = new EmailService();

File: python-backend/services/email_service.py
import msal
import requests
from datetime import datetime

class EmailService:
    def __init__(self, config):
        self.config = config
        self.client_id = config.get('OUTLOOK_CLIENT_ID')
        self.client_secret = config.get('OUTLOOK_CLIENT_SECRET')
        self.tenant_id = config.get('OUTLOOK_TENANT_ID')
        self.sender_email = config.get('OUTLOOK_SENDER_EMAIL', 'info@suniquecabinetry.com')
        
        if not self.client_id or not self.client_secret or not self.tenant_id:
            print('Outlook API credentials are not properly configured')
            self.is_configured = False
        else:
            self.is_configured = True
            self.authority = f"https://login.microsoftonline.com/{self.tenant_id}"
            self.scope = ['https://graph.microsoft.com/.default']
            
            self.app = msal.ConfidentialClientApplication(
                self.client_id,
                authority=self.authority,
                client_credential=self.client_secret
            )
    
    def get_access_token(self):
        """Get Microsoft Graph API access token"""
        result = self.app.acquire_token_silent(self.scope, account=None)
        
        if not result:
            result = self.app.acquire_token_for_client(scopes=self.scope)
        
        if "access_token" in result:
            return result['access_token']
        else:
            raise Exception(f"Failed to acquire token: {result.get('error_description', 'Unknown error')}")
    
    def send_confirmation_email(self, order_number, pickup_time, customer_email):
        """Send appointment confirmation email"""
        if not self.is_configured:
            print('Email service not configured. Skipping email send.')
            return {'success': False, 'message': 'Email service not configured'}
        
        try:
            if not customer_email:
                return {'success': False, 'message': 'Customer email is required'}
            
            # Generate email body
            email_body = self.generate_email_body(order_number, pickup_time)
            
            # Prepare email message
            message = {
                'message': {
                    'subject': f'Appointment Confirmation - Order {order_number}',
                    'body': {
                        'contentType': 'HTML',
                        'content': email_body
                    },
                    'toRecipients': [
                        {
                            'emailAddress': {
                                'address': customer_email
                            }
                        }
                    ],
                    'ccRecipients': [
                        {
                            'emailAddress': {
                                'address': 'zackwu204@gmail.com'
                            }
                        }
                    ]
                },
                'saveToSentItems': True
            }
            
            # Send email via Microsoft Graph API
            token = self.get_access_token()
            headers = {
                'Authorization': f'Bearer {token}',
                'Content-Type': 'application/json'
            }
            
            url = f'https://graph.microsoft.com/v1.0/users/{self.sender_email}/sendMail'
            response = requests.post(url, headers=headers, json=message)
            
            if response.status_code == 202:
                print(f'Confirmation email sent successfully to {customer_email}')
                return {
                    'success': True,
                    'message': 'Email sent successfully'
                }
            else:
                print(f'Failed to send email: {response.status_code} - {response.text}')
                return {
                    'success': False,
                    'message': f'Failed to send email: {response.status_code}'
                }
                
        except Exception as e:
            print(f'Error sending email: {e}')
            return {
                'success': False,
                'message': f'Error sending email: {str(e)}'
            }
    
    def generate_email_body(self, order_number, pickup_time):
        """Generate HTML email body"""
        try:
            # Parse pickup time
            pickup_date = datetime.fromisoformat(pickup_time.replace('Z', '+00:00'))
            formatted_date = pickup_date.strftime('%B %d, %Y')
            formatted_time = pickup_date.strftime('%I:%M %p')
            
        except:
            formatted_date = 'N/A'
            formatted_time = 'N/A'
        
        return f"""
        <!DOCTYPE html>
        <html>
        <head>
            <style>
                body {{
                    font-family: Arial, sans-serif;
                    line-height: 1.6;
                    color: #333;
                }}
                .container {{
                    max-width: 600px;
                    margin: 0 auto;
                    padding: 20px;
                }}
                .header {{
                    background-color: #8B4513;
                    color: white;
                    padding: 20px;
                    text-align: center;
                }}
                .content {{
                    background-color: #f9f9f9;
                    padding: 30px;
                    border-radius: 5px;
                    margin-top: 20px;
                }}
                .appointment-details {{
                    background-color: white;
                    padding: 20px;
                    border-left: 4px solid #8B4513;
                    margin: 20px 0;
                }}
                .detail-row {{
                    margin: 10px 0;
                }}
                .label {{
                    font-weight: bold;
                    color: #8B4513;
                }}
                .footer {{
                    margin-top: 30px;
                    padding-top: 20px;
                    border-top: 1px solid #ddd;
                    font-size: 14px;
                    color: #666;
                }}
            </style>
        </head>
        <body>
            <div class="container">
                <div class="header">
                    <h1>Appointment Confirmed</h1>
                </div>
                
                <div class="content">
                    <p>Dear Customer,</p>
                    
                    <p>Your pickup appointment has been successfully scheduled.</p>
                    
                    <div class="appointment-details">
                        <div class="detail-row">
                            <span class="label">Order Number:</span> {order_number}
                        </div>
                        <div class="detail-row">
                            <span class="label">Pickup Date:</span> {formatted_date}
                        </div>
                        <div class="detail-row">
                            <span class="label">Pickup Time:</span> {formatted_time}
                        </div>
                        <div class="detail-row">
                            <span class="label">Location:</span> Westgate Warehouse<br>
                            2045 Westgate Dr Ste 130, Carrollton, TX 75006
                        </div>
                    </div>
                    
                    <p><strong>Important Information:</strong></p>
                    <ul>
                        <li>Please arrive within your scheduled time slot</li>
                        <li>Bring your order number for reference</li>
                        <li>Contact us if you need to reschedule</li>
                    </ul>
                    
                    <div class="footer">
                        <p>If you have any questions, please contact us:</p>
                        <p>
                            <strong>Sunique Cabinetry</strong><br>
                            Phone: (972) 245-3309<br>
                            Email: info@suniquecabinetry.com<br>
                            Address: 2045 Westgate Dr Ste 130, Carrollton, TX 75006
                        </p>
                    </div>
                </div>
            </div>
        </body>
        </html>
        """
    
    def send_cancellation_email(self, order_number, original_pickup_time, customer_email):
        """Send appointment cancellation email"""
        if not self.is_configured:
            print('Email service not configured. Skipping email send.')
            return {'success': False, 'message': 'Email service not configured'}
        
        try:
            if not customer_email:
                return {'success': False, 'message': 'Customer email is required'}
            
            # Generate email body
            email_body = self.generate_cancellation_email_body(order_number, original_pickup_time)
            
            # Prepare email message
            message = {
                'message': {
                    'subject': f'Appointment Cancelled - Order {order_number}',
                    'body': {
                        'contentType': 'HTML',
                        'content': email_body
                    },
                    'toRecipients': [
                        {
                            'emailAddress': {
                                'address': customer_email
                            }
                        }
                    ],
                    'ccRecipients': [
                        {
                            'emailAddress': {
                                'address': 'zackwu204@gmail.com'
                            }
                        }
                    ]
                },
                'saveToSentItems': True
            }
            
            # Send email via Microsoft Graph API
            token = self.get_access_token()
            headers = {
                'Authorization': f'Bearer {token}',
                'Content-Type': 'application/json'
            }
            
            url = f'https://graph.microsoft.com/v1.0/users/{self.sender_email}/sendMail'
            response = requests.post(url, headers=headers, json=message)
            
            if response.status_code == 202:
                print(f'Cancellation email sent successfully to {customer_email}')
                return {
                    'success': True,
                    'message': 'Cancellation email sent successfully'
                }
            else:
                print(f'Failed to send cancellation email: {response.status_code} - {response.text}')
                return {
                    'success': False,
                    'message': f'Failed to send email: {response.status_code}'
                }
                
        except Exception as e:
            print(f'Error sending cancellation email: {e}')
            return {
                'success': False,
                'message': f'Error sending email: {str(e)}'
            }
    
    def send_reschedule_email(self, order_number, old_pickup_time, new_pickup_time, customer_email):
        """Send appointment reschedule email"""
        if not self.is_configured:
            print('Email service not configured. Skipping email send.')
            return {'success': False, 'message': 'Email service not configured'}
        
        try:
            if not customer_email:
                return {'success': False, 'message': 'Customer email is required'}
            
            # Generate email body
            email_body = self.generate_reschedule_email_body(order_number, old_pickup_time, new_pickup_time)
            
            # Prepare email message
            message = {
                'message': {
                    'subject': f'Appointment Rescheduled - Order {order_number}',
                    'body': {
                        'contentType': 'HTML',
                        'content': email_body
                    },
                    'toRecipients': [
                        {
                            'emailAddress': {
                                'address': customer_email
                            }
                        }
                    ],
                    'ccRecipients': [
                        {
                            'emailAddress': {
                                'address': 'zackwu204@gmail.com'
                            }
                        }
                    ]
                },
                'saveToSentItems': True
            }
            
            # Send email via Microsoft Graph API
            token = self.get_access_token()
            headers = {
                'Authorization': f'Bearer {token}',
                'Content-Type': 'application/json'
            }
            
            url = f'https://graph.microsoft.com/v1.0/users/{self.sender_email}/sendMail'
            response = requests.post(url, headers=headers, json=message)
            
            if response.status_code == 202:
                print(f'Reschedule email sent successfully to {customer_email}')
                return {
                    'success': True,
                    'message': 'Reschedule email sent successfully'
                }
            else:
                print(f'Failed to send reschedule email: {response.status_code} - {response.text}')
                return {
                    'success': False,
                    'message': f'Failed to send email: {response.status_code}'
                }
                
        except Exception as e:
            print(f'Error sending reschedule email: {e}')
            return {
                'success': False,
                'message': f'Error sending email: {str(e)}'
            }
    
    def generate_cancellation_email_body(self, order_number, original_pickup_time):
        """Generate HTML email body for cancellation"""
        try:
            # Parse pickup time
            pickup_date = datetime.fromisoformat(original_pickup_time.replace('Z', '+00:00'))
            formatted_date = pickup_date.strftime('%B %d, %Y')
            formatted_time = pickup_date.strftime('%I:%M %p')
            
        except:
            formatted_date = 'N/A'
            formatted_time = 'N/A'
        
        return f"""
        <!DOCTYPE html>
        <html>
        <head>
            <style>
                body {{
                    font-family: Arial, sans-serif;
                    line-height: 1.6;
                    color: #333;
                }}
                .container {{
                    max-width: 600px;
                    margin: 0 auto;
                    padding: 20px;
                }}
                .header {{
                    background-color: #dc3545;
                    color: white;
                    padding: 20px;
                    text-align: center;
                }}
                .content {{
                    background-color: #f9f9f9;
                    padding: 30px;
                    border-radius: 5px;
                    margin-top: 20px;
                }}
                .appointment-details {{
                    background-color: white;
                    padding: 20px;
                    border-left: 4px solid #dc3545;
                    margin: 20px 0;
                }}
                .detail-row {{
                    margin: 10px 0;
                }}
                .label {{
                    font-weight: bold;
                    color: #dc3545;
                }}
                .footer {{
                    margin-top: 30px;
                    padding-top: 20px;
                    border-top: 1px solid #ddd;
                    font-size: 14px;
                    color: #666;
                }}
            </style>
        </head>
        <body>
            <div class="container">
                <div class="header">
                    <h1>Appointment Cancelled</h1>
                </div>
                
                <div class="content">
                    <p>Dear Valued Customer,</p>
                    
                    <p>Your pickup appointment has been cancelled.</p>
                    
                    <div class="appointment-details">
                        <div class="detail-row">
                            <span class="label">Order Number:</span> {order_number}
                        </div>
                        <div class="detail-row">
                            <span class="label">Cancelled Appointment Date:</span> {formatted_date}
                        </div>
                        <div class="detail-row">
                            <span class="label">Cancelled Appointment Time:</span> {formatted_time}
                        </div>
                    </div>
                    
                    <p><strong>Need to Schedule a New Appointment?</strong></p>
                    <p>If you need to schedule a new pickup appointment, please use our online booking system or contact us directly.</p>
                    
                    <div class="footer">
                        <p>If you have any questions, please contact us:</p>
                        <p>
                            <strong>Sunique Cabinetry</strong><br>
                            Phone: (972) 245-3309<br>
                            Email: info@suniquecabinetry.com<br>
                            Address: 2045 Westgate Dr Ste 130, Carrollton, TX 75006
                        </p>
                    </div>
                </div>
            </div>
        </body>
        </html>
        """
    
    def generate_reschedule_email_body(self, order_number, old_pickup_time, new_pickup_time):
        """Generate HTML email body for reschedule"""
        try:
            # Parse old pickup time
            old_date = datetime.fromisoformat(old_pickup_time.replace('Z', '+00:00'))
            old_formatted_date = old_date.strftime('%B %d, %Y')
            old_formatted_time = old_date.strftime('%I:%M %p')
            
            # Parse new pickup time
            new_date = datetime.fromisoformat(new_pickup_time.replace('Z', '+00:00'))
            new_formatted_date = new_date.strftime('%B %d, %Y')
            new_formatted_time = new_date.strftime('%I:%M %p')
            
        except:
            old_formatted_date = 'N/A'
            old_formatted_time = 'N/A'
            new_formatted_date = 'N/A'
            new_formatted_time = 'N/A'
        
        return f"""
        <!DOCTYPE html>
        <html>
        <head>
            <style>
                body {{
                    font-family: Arial, sans-serif;
                    line-height: 1.6;
                    color: #333;
                }}
                .container {{
                    max-width: 600px;
                    margin: 0 auto;
                    padding: 20px;
                }}
                .header {{
                    background-color: #ffc107;
                    color: #333;
                    padding: 20px;
                    text-align: center;
                }}
                .content {{
                    background-color: #f9f9f9;
                    padding: 30px;
                    border-radius: 5px;
                    margin-top: 20px;
                }}
                .appointment-details {{
                    background-color: white;
                    padding: 20px;
                    border-left: 4px solid #ffc107;
                    margin: 20px 0;
                }}
                .old-time {{
                    text-decoration: line-through;
                    color: #999;
                }}
                .new-time {{
                    color: #28a745;
                    font-weight: bold;
                }}
                .detail-row {{
                    margin: 10px 0;
                }}
                .label {{
                    font-weight: bold;
                    color: #ffc107;
                }}
                .footer {{
                    margin-top: 30px;
                    padding-top: 20px;
                    border-top: 1px solid #ddd;
                    font-size: 14px;
                    color: #666;
                }}
            </style>
        </head>
        <body>
            <div class="container">
                <div class="header">
                    <h1>Appointment Rescheduled</h1>
                </div>
                
                <div class="content">
                    <p>Dear Valued Customer,</p>
                    
                    <p>Your pickup appointment has been rescheduled by our staff.</p>
                    
                    <div class="appointment-details">
                        <div class="detail-row">
                            <span class="label">Order Number:</span> {order_number}
                        </div>
                        <div class="detail-row">
                            <span class="label">Previous Time:</span><br>
                            <span class="old-time">{old_formatted_date} at {old_formatted_time}</span>
                        </div>
                        <div class="detail-row">
                            <span class="label">New Time:</span><br>
                            <span class="new-time">{new_formatted_date} at {new_formatted_time}</span>
                        </div>
                        <div class="detail-row">
                            <span class="label">Location:</span> Westgate Warehouse<br>
                            2045 Westgate Dr Ste 130, Carrollton, TX 75006
                        </div>
                    </div>
                    
                    <p><strong>Important Information:</strong></p>
                    <ul>
                        <li>Please arrive within your new scheduled time slot</li>
                        <li>Bring your order number for reference</li>
                        <li>Contact us if you have any questions</li>
                    </ul>
                    
                    <div class="footer">
                        <p>If you have any questions, please contact us:</p>
                        <p>
                            <strong>Sunique Cabinetry</strong><br>
                            Phone: (972) 245-3309<br>
                            Email: info@suniquecabinetry.com<br>
                            Address: 2045 Westgate Dr Ste 130, Carrollton, TX 75006
                        </p>
                    </div>
                </div>
            </div>
        </body>
        </html>
        """

